From: Mahesh Salgaonkar <mahesh@linux.vnet.ibm.com>
Date: Fri, 11 Mar 2011 14:03:10 +0900
Subject: [PATCH] Copy correct nr_cpus info to dumpfile during re-filtering.

During re-filtering nr_cpus is to 0 in the o/p dumpfile and hence crash
utility fails to open the re-filtered dumpfile.

This patch fixes this issue by copying nr_cpus info from kdump-compressed
dumpfile to info->nr_cpus which then get's correctly copied to o/p
dumpfile.

Signed-off-by: Mahesh Salgaonkar <mahesh@linux.vnet.ibm.com>
(cherry picked from commit c4f1c98a9827c1c0e41772c1954940fbf1b48048)
---
 makedumpfile.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/makedumpfile.c b/makedumpfile.c
index a0fad6a..7d0c663 100644
--- a/makedumpfile.c
+++ b/makedumpfile.c
@@ -924,6 +924,7 @@ get_kdump_compressed_header_info(char *filename)
 		goto error;
 	}
 	memcpy(info->kh_memory, &kh, sizeof(kh));
+	info->nr_cpus = dh.nr_cpus;
 
 	if (dh.header_version >= 3) {
 		/* A dumpfile contains vmcoreinfo data. */
-- 
